rocketpy/simulation/flight_comparator.py

```diff
@@ -0,0 +1,619 @@
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from rocketpy.mathutils import Function
+
+from ..plots.plot_helpers import show_or_save_fig
+
+
+# pylint: disable=too-many-public-methods
+# pylint: disable=too-many-statements
+class FlightComparator:
+    """
+    A class to compare a RocketPy Flight simulation against external data sources
+    (such as flight logs, OpenRocket simulations, RAS Aero).
+
+    This class handles the time-interpolation required to compare datasets
+    recorded at different frequencies.
+
+    Parameters
+    ----------
+    flight : Flight
+        The reference RocketPy Flight object to compare against.
+
+    Attributes
+    ----------
+    flight : Flight
+        The reference RocketPy Flight object to compare against.
+    data_sources : dict
+        Dictionary storing external data sources in the format
+        {'Source Name': {'variable': Function}}.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        # Assuming you have a Flight object named 'my_flight'
+        comparator = FlightComparator(my_flight)
+
+        # Add external data
+        comparator.add_data('OpenRocket', {
+            'altitude': (time_array, altitude_array),
+            'vz': (time_array, velocity_array)
+        })
+
+        # Run comparisons
+        comparator.compare('altitude')
+        comparator.summary()
+        events_table = comparator.compare_key_events()
+    """
+
+    def __init__(self, flight):
+        """
+        Initialize the comparator with a reference RocketPy Flight.
+
+        Parameters
+        ----------
+        flight : rocketpy.Flight
+            The reference RocketPy Flight object to compare against.
+
+        Returns
+        -------
+        None
+        """
+        self.flight = flight
+        self.data_sources = {}  # The format is {'Source Name': {'variable': Function}}
+
+    def add_data(self, label, data_dict):
```

Comment:
question: What do you thikn about acepting a `Flight` object here?

Reply:
I need to think baout it for a while. Gonna get back to you soon.

usually when these things happen I try to investigate whether there are similar behaviors throughout the project.

Reply:
@Monta120 I carefully reviewed it and I truly believe the best option here is for us to remove the "convinient wrapper" from the Flight class. This approach would remove the circular imports problem, with the only downside of requiring 1-2 extra lines from the user. I believe it is worth the changes.